pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -156,36 +161,33 @@ func (c *Cluster) Start() error {
 			if err := updateDeploymentAndWait(c.context, c.clusterInfo, d, config.MgrType, mgrConfig.DaemonID, c.spec.SkipUpgradeChecks, false); err != nil {
 				logger.Errorf("failed to update mgr deployment %q. %v", resourceName, err)
 			}
+		} else {
+			// wait for the new deployment
+			deploymentsToWaitFor = append(deploymentsToWaitFor, newDeployment)
 		}
 	}
 
-	if err := c.configureDashboardService(); err != nil {
-		logger.Errorf("failed to enable dashboard. %v", err)
+	// If the mgr is newly created, wait for it to start before continuing with the service and
+	// module configuration
+	for _, d := range deploymentsToWaitFor {
```

Comment:
When did we have that conversation? :) It's really independent of the multiple mgr problem, but it will be a good thing for stability of the mgr module initialization.

Reply:
I think I had mentioned that in https://github.com/rook/rook/pull/7235#discussion_r577538145